src/classes/ADDR_Addresses_TDTM.cls

```diff
@@ -67,231 +82,231 @@ public class ADDR_Addresses_TDTM extends TDTM_Runnable {
+	        // - the parent Account, if not null
+	        // - the parent Contact, if not null
+	        // - any other Addresses that should no longer be Default
+	        
+	        // map of parents, and their default Address
+	        Map<Id, Address__c> mapParentIdAddr = new Map<Id, Address__c>();
 	            
-	            // BEFORE INSERT
-	            if (triggerAction == TDTM_Runnable.Action.BeforeInsert) {
-	               // when an address is marked Default, we update its latest date fields
-	               if (addr.Default_Address__c) { 
-	                   addr.Latest_Start_Date__c = System.today();
-	                   addr.Latest_End_Date__c = null;
+	        // BEFORE INSERT
+	        if (!alreadyRunBeforeInsert && triggerAction == TDTM_Runnable.Action.BeforeInsert) {
+		        for (SObject so : listNew) {
+		            Address__c addr = (Address__c)so;
+	                // when an address is created as Default, we update its latest date fields
+	                if (addr.Default_Address__c) {
+	                    addr.Latest_Start_Date__c = System.today();
+	                    addr.Latest_End_Date__c = null;
 	               }
-	            }
-	            
-	            // BEFORE UPDATE
-	            if (triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
+		        }
+		        alreadyRunBeforeInsert = true;
+	        }
+	        
+	        // BEFORE UPDATE
+	        if (!alreadyRunBeforeUpdate && triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
+	            Integer i = 0;
+		        for (SObject so : listNew) {
+		            Address__c addr = (Address__c)so;
+		            Address__c addrOld = listOld == null? null : (Address__c)listOld[i];
 	                if (addr.Default_Address__c != addrOld.Default_Address__c) {
-	                   // when an address changes Default, we update its latest date fields
+	                   // when a Default address changes, we update its latest date fields
 	                    if (addr.Default_Address__c) {
 	                       addr.Latest_Start_Date__c = System.today();
 	                       addr.Latest_End_Date__c = null;
 	                    } else {
+	                       UTIL_Debug.debug('****Address updated to not default');
 	                       addr.Latest_End_Date__c = System.today();                        
 	                    }
 	                }
+	                i++;
+		        }
+		        alreadyRunBeforeUpdate = true;
+	        }
+	    
+	        // AFTER INSERT
+	        if (!alreadyRunAfterInsert && triggerAction == TDTM_Runnable.Action.AfterInsert) {
+	            for (SObject so : listNew) {
+	                Address__c addr = (Address__c)so;
+		            // a new address that is marked default or seasonal needs to propogate to the parent Account
+		            if (addr.Default_Address__c || ADDR_Seasonal_UTIL.isSeasonalAddr(addr)) {
```

Comment:
I think this is not a problem, because we are just marking it as "need to look into it and possibly update". This ends up calling the [storeDefaultSeasonal](https://github.com/SalesforceFoundation/HEDAP/blob/feature/245-contact-addresses/src/classes/ADDR_Addresses_TDTM.cls#L357) method that will determine if a Seasonal address is in effect.